File: centsible-api/src/main/kotlin/beer/thierry/centsible/api/model/transaction/TransactionFilters.kt
package beer.thierry.centsible.api.model.transaction

import beer.thierry.centsible.api.model.category.CategoryType
import java.time.LocalDate

enum class TransactionSort {
    DATE_DESC, DATE_ASC, AMOUNT_DESC, AMOUNT_ASC
}

data class TransactionFilters(
    val search: String? = null,
    val categoryIds: List<Long>? = null,
    val from: LocalDate? = null,
    val to: LocalDate? = null,
    val type: CategoryType? = null,
    val sort: TransactionSort = TransactionSort.DATE_DESC,
)

File: centsible-ui/app/models/transactions/transaction-filters.ts
export type TransactionSort = 'DATE_DESC' | 'DATE_ASC' | 'AMOUNT_DESC' | 'AMOUNT_ASC';

export type TransactionTypeFilter = 'INCOME' | 'EXPENSE';

export interface TransactionFilters {
  search?: string;
  categoryIds?: number[];
  tagIds?: number[];
  fromDate?: string;
  toDate?: string;
  type?: TransactionTypeFilter;
  amountMin?: number;
  amountMax?: number;
  sort?: TransactionSort;
}

export interface CategoryDrillPayload {
  categoryId: number;
  categoryName: string;
  fromDate: string | null;
  toDate: string | null;
}
